pylint/test/unittest_checker_spelling.py

```diff
@@ -76,3 +104,101 @@ def test_invalid_docstring_characters(self):
+
+    @pytest.mark.skipif(spell_dict is None,
+                        reason="missing python-enchant package or missing "
+                        "spelling dictionaries")
+    @set_config(spelling_dict=spell_dict)
+    def test_skip_top_level_pylint_enable_disable_comments(self):
+        self.checker.process_tokens(tokenize_str('# Line 1\n Line 2\n# pylint: disable=ungrouped-imports'))
+        assert self.linter.release_messages() == []
+
+    @pytest.mark.skipif(spell_dict is None,
+                        reason="missing python-enchant package or missing "
+                        "spelling dictionaries")
+    @set_config(spelling_dict=spell_dict)
+    def test_skip_words_with_numbers(self):
+        self.checker.process_tokens(tokenize_str('\n# 0ne\n# Thr33\n# Sh3ll'))
+        assert self.linter.release_messages() == []
+
+    @pytest.mark.skipif(spell_dict is None,
+                        reason="missing python-enchant package or missing "
+                        "spelling dictionaries")
+    @set_config(spelling_dict=spell_dict)
+    def test_skip_camel_cased_words(self):
+        stmt = astroid.extract_node(
+            'class ComentAbc(object):\n   """ComentAbc with a bad coment"""\n   pass')
+        try:
+            with self.assertAddsMessages(
+                Message('wrong-spelling-in-docstring', line=2,
+                        args=('coment', 'ComentAbc with a bad coment',
+                              '                     ^^^^^^',
+                              "comet' or 'comment' or 'moment' or 'foment"))):
+                self.checker.visit_classdef(stmt)
+        except AssertionError:
+            # In Arch Linux, at least, the suggestions do not match
+            with self.assertAddsMessages(
+                Message('wrong-spelling-in-docstring', line=2,
+                        args=('coment', 'ComentAbc with a bad coment',
+                              '                     ^^^^^^',
+                              "comet' or 'comment' or 'cement' or 'cogent"))):
+                self.checker.visit_classdef(stmt)
+
+    @pytest.mark.skipif(spell_dict is None,
+                        reason="missing python-enchant package or missing "
+                        "spelling dictionaries")
+    @set_config(spelling_dict=spell_dict)
+    def test_skip_words_with_underscores(self):
+        stmt = astroid.extract_node(
+            'def fff(param_name):\n   """test param_name"""\n   pass')
+        self.checker.visit_functiondef(stmt)
+        assert self.linter.release_messages() == []
+
+    @pytest.mark.skipif(spell_dict is None,
+                        reason="missing python-enchant package or missing "
+                        "spelling dictionaries")
+    @set_config(spelling_dict=spell_dict)
+    def test_skip_email_address(self):
+        self.checker.process_tokens(tokenize_str('# uname@domain.tld'))
+        assert self.linter.release_messages() == []
+
+    @pytest.mark.skipif(spell_dict is None,
```

Comment:
maybe declare your custom decorator in global scope, `skip_if_enchant_unavilable = pytest.mark.skipif(spell_dict is None, reason="missing python-enchant package or missing spelling dictionaries")` and use it to decorate all of methods. There would be slightly less copy-pasted code all around